rclcpp_action/test/test_server.cpp

```diff
@@ -32,15 +33,609 @@ class TestServer : public ::testing::Test
   {
     rclcpp::init(0, nullptr);
   }
+
+  std::shared_ptr<test_msgs::action::Fibonacci::GoalRequestService::Request>
+  send_goal_request(rclcpp::Node::SharedPtr node, std::array<uint8_t, 16> uuid)
+  {
+    auto client = node->create_client<test_msgs::action::Fibonacci::GoalRequestService>(
+      "fibonacci/_action/send_goal");
+    if (!client->wait_for_service(std::chrono::seconds(20))) {
+      throw std::runtime_error("send goal service didn't become available");
+    }
+    auto request = std::make_shared<test_msgs::action::Fibonacci::GoalRequestService::Request>();
+    request->uuid = uuid;
+    auto future = client->async_send_request(request);
+    if (rclcpp::executor::FutureReturnCode::SUCCESS !=
+      rclcpp::spin_until_future_complete(node, future))
+    {
+      throw std::runtime_error("send goal future didn't complete succesfully");
+    }
+    return request;
+  }
+
+  void
+  send_cancel_request(rclcpp::Node::SharedPtr node, std::array<uint8_t, 16> uuid)
+  {
+    auto cancel_client = node->create_client<test_msgs::action::Fibonacci::CancelGoalService>(
+      "fibonacci/_action/cancel_goal");
+    if (!cancel_client->wait_for_service(std::chrono::seconds(20))) {
+      throw std::runtime_error("cancel goal service didn't become available");
+    }
+    auto request = std::make_shared<test_msgs::action::Fibonacci::CancelGoalService::Request>();
+    request->goal_info.goal_id.uuid = uuid;
+    auto future = cancel_client->async_send_request(request);
+    if (rclcpp::executor::FutureReturnCode::SUCCESS !=
+      rclcpp::spin_until_future_complete(node, future))
+    {
+      throw std::runtime_error("cancel goal future didn't complete succesfully");
+    }
+  }
 };
 
 TEST_F(TestServer, construction_and_destruction)
 {
-  auto node = std::make_shared<rclcpp::Node>("my_node", "/rclcpp_action/test/server");
+  auto node = std::make_shared<rclcpp::Node>("construct_node", "/rclcpp_action/construct");
 
   using GoalHandle = rclcpp_action::ServerGoalHandle<test_msgs::action::Fibonacci>;
-  auto as = rclcpp_action::create_server<test_msgs::action::Fibonacci>(node.get(), "fibonacci",
-      [](rcl_action_goal_info_t &, test_msgs::action::Fibonacci::Goal *) {},
+  auto as = rclcpp_action::create_server<test_msgs::action::Fibonacci>(node, "fibonacci",
```

Comment:
We could add tests for invalid constructions (e.g. invalid node or action name).